packages/main/src/Tokenizer.ts

```diff
@@ -211,16 +266,27 @@ class Tokenizer extends UI5Element {
 		ResizeHandler.deregister(this.contentDom, this._resizeHandler);
 	}
 
-	async _openMorePopoverAndFireEvent() {
+	async _handleNMoreClick() {
+		if (this.disabled) {
+			return;
+		}
+
+		this._openedByNmore = true;
+		this.expanded = true;
+
 		if (!this.preventPopoverOpen) {
 			await this.openMorePopover();
+			this.scrollToEnd();
 		}
 
 		this.fireEvent("show-more-items-press");
 	}
 
 	async openMorePopover() {
-		(await this.getPopover()).showAt(this.morePopoverOpener || this);
+		// the morePopoverProperty is an object so it will always return 'true', so we check for keys
+		const popoverOpener = Object.keys(this.morePopoverOpener).length === 0 ? this : this.morePopoverOpener;
+
+		(await this.getPopover()).showAt(popoverOpener);
 	}
 
 	_getTokens() {
```

Comment:
// same getter twice? decide which one to keep

Reply:
Done.